cql3/statements/create_keyspace_statement.cc

```diff
@@ -94,21 +94,42 @@ future<std::tuple<::shared_ptr<cql_transport::event::schema_change>, std::vector
     const auto& feat = qp.proxy().features();
     ::shared_ptr<event::schema_change> ret;
     std::vector<mutation> m;
+    std::vector<sstring> warnings;
 
     try {
-        m = service::prepare_new_keyspace_announcement(qp.db().real_database(), _attrs->as_ks_metadata(_name, tm, feat), ts);
+        auto ksm = _attrs->as_ks_metadata(_name, tm, feat);
+        m = service::prepare_new_keyspace_announcement(qp.db().real_database(), ksm, ts);
 
         ret = ::make_shared<event::schema_change>(
                 event::schema_change::change_type::CREATED,
                 event::schema_change::target_type::KEYSPACE,
                 keyspace());
+
+        // If the new keyspace uses tablets, as long as there are features
+        // which aren't supported by tablets we want to warn the user that
+        // they will not be usable on the new keyspace - and suggest how a
+        // keyspace can be created without tablets (see rationale in #16807).
+        // Once all feature will become supported with tablets, we should
+        // remove this check.
+        auto rs = locator::abstract_replication_strategy::create_replication_strategy(
+            ksm->strategy_name(),
+            locator::replication_strategy_params(ksm->strategy_options(), ksm->initial_tablets()));
+        if (rs->uses_tablets()) {
+            warnings.push_back(
+                "Tables in this keyspace will be replicated using tablets, "
```

Comment:
"Tables ... tablets..." - these 2 words are so similar, it's confusing. Since it's a message shown to the client, maybe we should emphasize what we really mean. But that's just a food for thought.

Reply:
I know, and I often fall on this confusion myself, but this is the word that Avi decided to use.... "vnodes" was a better word in the sense it doesn't mean anything or sound like anything else. So I don't know how I can improve this.

Reply:
Yeah, me neither, I guess we have to learn to live with that.